src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -702,7 +722,11 @@ static char* get_user_name(int vmid, int *nspid, TRAPS) {
 //
 static char* get_sharedmem_filename(const char* dirname, int vmid, int nspid) {
 
+#if defined(LINUX)
   int pid = (nspid == -1) ? vmid : nspid;
+#else
+  int pid = vmid;
+#endif
```

Comment:
This would be cleaner with LINUX_ONLY/NOT_L:INUX macros.

Reply:
done